third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -214,14 +225,40 @@ static void moveUpTranspose(triton::FuncOp funcOp) {
 }
 
 // Schedule global load and local store ops for better GEMM performance.
-static void scheduleGlobalLoadLocalStore(triton::FuncOp funcOp) {
+static void scheduleGlobalLoadLocalStore(scf::ForOp forOp) {
   SmallVector<Operation *> moveOps;
-  // Move local_stores early if dependence distance greater than one iteration.
-  // Best perf on GEMM when these precede global loads.
-  funcOp.walk([&](ttg::LocalStoreOp op) { moveOps.push_back(op); });
-  // Move global loads early to prefetch. This may increase register pressure
-  // but it enables issuing global loads early.
-  funcOp.walk([&](triton::LoadOp op) { moveOps.push_back(op); });
+  // Search through the forOp initArgs to find global loads and local stores for
+  // a GEMM that the pipeliner may have peeled into a loop prologue.
+  SmallVector<Value> vals = forOp.getInitArgs();
+  while (!vals.empty()) {
+    SmallVector<Value> nextVals; // next set of values to search via BFS.
+    for (size_t i = 0; i < vals.size(); ++i) {
+      Operation *defOp = vals[i].getDefiningOp();
+      if (defOp != nullptr && isa<triton::LoadOp>(defOp)) {
+        moveOps.push_back(defOp);
+      } else {
+        for (Operation *op : vals[i].getUsers()) {
```

Comment:
Don't need to capture these since they will be dropped below (see `leadsToLoad`).

But still needs to recurse..

Reply:
Done